doc/release/upcoming_changes/23528.compatibility.rst

```diff
@@ -0,0 +1,15 @@
+Default build against NumPy is now backwards compatible
```

Comment:
This section header may need a tweak to avoid reader confusion, maybe "Building against the NumPy C API now yields binaries that work with older NumPy versions"